druid/src/widget/images.rs

```diff
@@ -0,0 +1,254 @@
+            phantom: Default::default(),
+            fill: FillStrat::default(),
+        }
+    }
+
+    /// Change the fill stratacy of a newly constructed Image Widget
+    pub fn fill_mode(mut self, mode: FillStrat) -> Self {
+        self.fill = mode;
+        self
+    }
+
+    /// Change the fill stratagy of a Image Widgit inplace
+    pub fn set_fill(&mut self, newfil: FillStrat) {
+        self.fill = newfil;
+    }
+}
+
+impl<T: Data> Widget<T> for Image<T> {
+    fn event(&mut self, _ctx: &mut EventCtx, _event: &Event, _data: &mut T, _env: &Env) {}
+
+    fn lifecycle(&mut self, _ctx: &mut LifeCycleCtx, _event: &LifeCycle, _data: &T, _env: &Env) {}
+
+    fn update(&mut self, _ctx: &mut UpdateCtx, _old_data: &T, _data: &T, _env: &Env) {}
+
+    fn layout(
+        &mut self,
+        _layout_ctx: &mut LayoutCtx,
+        bc: &BoxConstraints,
+        _data: &T,
+        _env: &Env,
+    ) -> Size {
+        bc.debug_check("Image");
+
+        if bc.is_width_bounded() {
+            bc.max()
+        } else {
+            bc.constrain(self.image_data.get_size())
+        }
+    }
+    fn paint(&mut self, paint_ctx: &mut PaintCtx, _data: &T, _env: &Env) {
+        let offset_matrix =
+            get_affine_from_fill(paint_ctx.size(), self.image_data.get_size(), self.fill);
+
+        // The ImageData's to_piet function does not clip to the image's size
+        // CairoRenderContext is very like druids but with some extra goodies like clip
+        if self.fill == FillStrat::Contain {
+        } else {
+            let clip_rect = Rect::ZERO.with_size(paint_ctx.size());
+            paint_ctx.clip(clip_rect);
+        }
+        self.image_data.to_piet(offset_matrix, paint_ctx);
+    }
+}
+
+/// Stored Image data.
+/// Implements `FromStr` and can be converted to piet draw instructions.
+#[derive(Clone)]
+pub struct ImageData {
+    pixels: Vec<u8>,
+    x_pixels: u32,
+    y_pixels: u32,
+}
+
+impl ImageData {
+    /// Create an empty Image
+    pub fn empty() -> Self {
+        ImageData {
+            pixels: [].to_vec(),
+            x_pixels: 0,
+            y_pixels: 0,
+        }
+    }
+
+    /// This takes the raw binary contents of a file, eg png, jpg and returns a image
+    ///
+    /// If the image crate cant decode a image from the data then a error will be returned
+    pub fn from_data(raw_image: &[u8]) -> Result<Self, Box<dyn Error>> {
+        let dec = image::load_from_memory(raw_image).map_err(|e| e)?.to_rgb();
+
+        let sizeofimage = dec.dimensions();
+        Ok(ImageData {
+            pixels: dec.to_vec(),
+            x_pixels: sizeofimage.0,
+            y_pixels: sizeofimage.1,
+        })
+    }
+
+    /// Get the size in pixels of the contained image.
+    fn get_size(&self) -> Size {
+        Size::new(self.x_pixels as f64, self.y_pixels as f64)
+    }
+
+    /// Convert ImageData into Piet draw instructions
+    pub fn to_piet(&self, offset_matrix: Affine, paint_ctx: &mut PaintCtx) {
```

Comment:
does this actually need to be `pub`? Do we anticipate this being used outside of this widget?